lints.rs

```diff
@@ -13,36 +13,45 @@
     clippy::exit,
     clippy::explicit_into_iter_loop,
     clippy::filter_map_next,
+    clippy::float_cmp_const,
     clippy::fn_params_excessive_bools,
+    clippy::get_unwrap,
     clippy::if_let_mutex,
     clippy::imprecise_flops,
     clippy::inefficient_to_string,
     clippy::let_unit_value,
     clippy::linkedlist,
     clippy::lossy_float_literal,
     clippy::macro_use_imports,
+    clippy::map_err_ignore,
     clippy::map_flatten,
     clippy::map_unwrap_or,
     clippy::match_on_vec_items,
     clippy::match_wildcard_for_single_variants,
     clippy::mem_forget,
     clippy::mismatched_target_os,
+    clippy::mutex_integer,
     clippy::needless_borrow,
     clippy::needless_continue,
     clippy::option_option,
+    clippy::print_stdout,
```

Comment:
have tested with `print_stdout` and `print_stderr` lints but for some reason it wasn't detecting many of our `println!` and `eprintln!` calls which was surprising. need more investigating and testing before being comfortable / useful enabling it.

Did it seem to work alright for y'all?

Reply:
Indeed it strangely doesn't seem to propagate into some modules unless manually placing `#![warn]` on top or `#[warn]` on the `pub mod`. I wonder how many lints we are loosing this way, of if just `print_std*` is affected...

Reply:
Seems like it may be this bug: https://github.com/rust-lang/rust-clippy/issues/5721